File: silbiocomp/Practicals/Solutions/get_TreeHeight.py
import math

def tree_height(degrees, distance):
    """ Returns the height of a tree given a distance from its base 
				and a measured angle to its top."""
    radians = degrees * math.pi / 180
    height = distance * math.tan(radians)
    return height

# Data are in the form: ID of individual, distance (m) and angle (degrees)
trees = [ ('A', 31.66, 41.28),
          ('B', 45.98, 44.53),
          ('C', 31.24, 25.14),
          ('D',	34.61, 23.33),
          ('E', 45.46, 38.34),
          ('F', 48.79, 33.59),
          ('H', 30.64, 29.66),
         ]

tree_height(trees)

# For each individual, print its ID and height

File: silbiocomp/Practicals/Solutions/get_TreeHeight.R
# This script calculates heights of trees using a csv file as input 
# The csv file should contain a list tree species name, of distance of 
# each tree from its base and angle to its top.
#
# Heights are calculated using the trigonometric formula 
#
# height = distance * tan(radians)
#
# ARGUMENTS:
# degrees      The angle of elevation in radians
# distance     The distance from base (e.g., meters)
#
# OUTPUT:
# csv file with the heights of the tree in same units as "distance"

TreeHeight <- function(degrees, distance)
  {radians <- degrees * pi / 180
  height <- distance * tan(radians)
  return (height)
}

arg1 <- commandArgs(TRUE) # get file name input-ed from command line 
Trees <- read.csv(arg1[1])

Tree.Height.m <- TreeHeight(distance = Trees[,2], degrees = Trees[,3]) # applies function to column data and saves it as vector
Trees[,"Tree.Height.m"] <- Tree.Height.m # add vector to data frame

#Output new data frame as a csv file.
write.csv(Trees, "../Results/treeheights.csv",row.names=FALSE)
